homeassistant/components/mobile_app/websocket_api.py

```diff
@@ -0,0 +1,143 @@
+
+    device = hass.data[DOMAIN][DATA_REGISTRATIONS].get(webhook_id)
+
+    if device is None:
+        connection.send_error(msg['id'], ERR_NOT_FOUND,
+                              "Webhook ID not found in storage")
+        return
+
+    if device[CONF_USER_ID] != user.id and not user.is_admin:
+        return error_message(
+            msg['id'], ERR_UNAUTHORIZED, 'User is not registration owner')
+
+    connection.send_message(
+        result_message(msg['id'], safe_device(device)))
+
+
+@ws_require_user()
+@async_response
+@websocket_command({
+    vol.Required('type'): 'mobile_app/get_user_registrations',
+    vol.Optional(CONF_USER_ID): cv.string,
+})
+async def websocket_get_user_registrations(
+        hass: HomeAssistantType, connection: ActiveConnection,
+        msg: dict) -> None:
+    """Return all registrations or just registrations for given user ID."""
+    user_id = msg.get(CONF_USER_ID, connection.user.id)
+
+    if user_id != connection.user.id and not connection.user.is_admin:
+        # If user ID is provided and is not current user ID and current user
+        # isn't an admin user
+        connection.send_error(msg['id'], ERR_UNAUTHORIZED, "Unauthorized")
+        return
+
+    user_devices = []
+
+    for device in hass.data[DOMAIN][DATA_REGISTRATIONS].values():
+        if connection.user.is_admin or device[CONF_USER_ID] is user_id:
+            user_devices.append(safe_device(device))
+
+    connection.send_message(
+        result_message(msg['id'], user_devices))
+
+
+@ws_require_user()
+@async_response
+@websocket_command({
+    vol.Required('type'): 'mobile_app/delete_registration',
+    vol.Required(CONF_WEBHOOK_ID): cv.string,
+})
+async def websocket_delete_registration(hass: HomeAssistantType,
+                                        connection: ActiveConnection,
+                                        msg: dict) -> None:
+    """Delete the registration for the given webhook_id."""
+    user = connection.user
+
+    webhook_id = msg.get(CONF_WEBHOOK_ID)
+    if webhook_id is None:
+        connection.send_error(msg['id'], ERR_INVALID_FORMAT,
+                              "Webhook ID not provided")
+        return
+
+    device = hass.data[DOMAIN][DATA_REGISTRATIONS].get(webhook_id)
+
+    if device is None:
+        connection.send_error(msg['id'], ERR_NOT_FOUND,
+                              "Webhook ID not found in storage")
+        return
+
+    if device[CONF_USER_ID] != user.id and not user.is_admin:
+        return error_message(
+            msg['id'], ERR_UNAUTHORIZED, 'User is not registration owner')
+
+    del hass.data[DOMAIN][DATA_REGISTRATIONS][webhook_id]
+
+    hass.data[DOMAIN][DATA_DELETED_IDS].append(webhook_id)
+
+    store = hass.data[DOMAIN][DATA_STORE]
+
+    try:
+        await store.async_save(savable_state(hass))
+    except HomeAssistantError:
+        return error_message(
+            msg['id'], 'internal_error', 'Error deleting device')
+
+    if async_is_logged_in(hass):
```

Comment:
We should not have to check if logged in, we should only have to check if `"cloud" in hass.config.components` and only call this if registration contained a cloudhook url